demo/src/app/components/alert/demos/selfclosing/alert-selfclosing.ts

```diff
@@ -10,7 +10,7 @@ export class NgbdAlertSelfclosing implements OnInit {
   private _success = new Subject<string>();
 
   staticAlertClosed = false;
-  successMessage: string;
+  successMessage: string | null = null;
```

Comment:
Yes, it can be a good trick. 

At least, it's quite safe the demo part. We need to check case by case to see if a difference between empty string and null. But the code visibility/complexity seems better to me with empty string.

Reply:
+1 from me